apps/full-site-editing/full-site-editing-plugin/full-site-editing/plugins/editor-template-classes/index.js

```diff
@@ -0,0 +1,56 @@
+/* global fullSiteEditing */
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import { endsWith, get, map } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { withSelect } from '@wordpress/data';
+import { registerPlugin } from '@wordpress/plugins';
+
+const EditorTemplateClasses = withSelect( select => {
+	const { getEntityRecord } = select( 'core' );
+	const { getEditedPostAttribute } = select( 'core/editor' );
+	const templatePartClasses = map( getEditedPostAttribute( 'template_part_types' ), typeId => {
+		const typeName = get(
+			getEntityRecord( 'taxonomy', 'wp_template_part_type', typeId ),
+			'name',
+			''
+		);
+		if ( endsWith( typeName, '-header' ) ) {
+			return 'site-header site-branding';
+		}
+		if ( endsWith( typeName, '-footer' ) ) {
+			return 'site-footer';
+		}
+	} );
+	return { templatePartClasses };
+} )( ( { templatePartClasses } ) => {
+	const blockListInception = setInterval( () => {
+		const blockList = document.querySelector(
+			'.editor-block-list__layout.block-editor-block-list__layout'
```

Comment:
Since this chooses the first matching element from the dom, we're totally fine, but I came up against this in some other work so I figured I'd mention it juuust in case anything comes up 🤓

Nested blocks get wrapped by an element with these classes:
<img width="644" alt="Screen Shot 2019-07-31 at 9 27 04 AM" src="https://user-images.githubusercontent.com/6265975/62229942-a4dcf380-b375-11e9-976f-4aba82549cf1.png">

Reply:
Oh nice catch, and I'm glad I've accidentally avoided that bug! 😄